airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -1274,6 +1285,11 @@ def create_concurrent_cursor_from_datetime_based_cursor(
                 f"Expected manifest component of type {model_type.__name__}, but received {component_type} instead"
             )
 
+        # TODO validate and explain why we need to do this...
```

Comment:
agree, i don't think i fully understand why we're loading parameters into different keys, i thought the pydantic model parsing handles this